### DISCUSSION TITLE
Test accuracy consistently higher than Train_accuracy when used full dataset of 750 images in transfer learning

### DISCUSSION THREAD

**Discussion by NietzscheDostoevsky** [mrdbourke/pytorch-deep-learning#458] [General]
I used the create dataset notebook to create a full dataset of 750 images per class of Pizza, Sushi and Steak, and trained the same model for 20 epochs. I am getting test_accuracy consistently higher than train_accuracy, and test_loss lower than train_loss as well. 
What could be the reason for this ? 

![e_20_full_dataset](https://github.com/mrdbourke/pytorch-deep-learning/assets/96907475/92392435-95d4-4e8e-a219-db0634a1efde)

**Comment by mrdbourke** [comment]
Hi @DietzscheNostoevsky,

It may be because the patterns learned in the training data are *really* good at predicting on unseen images.

Have you tried printing out some outputs of the testing images with their predictions to see what they look like?

**Reply by NietzscheDostoevsky** [reply]
I did generated the confusion Matrix as given the exercise. This is what is looks like : 

![confusion_750images](https://github.com/mrdbourke/pytorch-deep-learning/assets/96907475/d101e36f-b78e-4e43-b736-1b50719736dd)

**Reply by mrdbourke** [reply]
That looks fantastic! Looks like your model is performing really well. 

Next I'd try to figure out where the model went wrong and see what the outputs look like.

If test accuracy is higher than training accuracy, it means your model may be underfitting. So some things you can do: 
* Train for longer
* Increase model capacity (more parameters/bigger model)
* Increase data inputs

**Reply by NietzscheDostoevsky** [reply]
You were right, the model was underfitting. I trained the model for more epochs (200 and 1000 ), and train loss was more than test loss 
200 epoch curve 
![e_200_full_dataset](https://github.com/mrdbourke/pytorch-deep-learning/assets/96907475/cf4ddce3-5295-4ce7-8967-8bb167917430)

1000 epoch curve
![e_1000_full_dataset](https://github.com/mrdbourke/pytorch-deep-learning/assets/96907475/11fa5727-04bf-4f78-bdbe-7144e50f8fa9)

Thank you so much for your guidance

**Reply by mrdbourke** [reply]
Nice work! Glad you figured it out and tried training the model for longer! Now it looks like your model is slightly overfitting :D
